src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -46,10 +48,22 @@ public void storeMessage(Message message) {
     messageEntity.setProperty("text", message.getText());
     messageEntity.setProperty("timestamp", message.getTimestamp());
     messageEntity.setProperty("sentimentScore", message.getSentimentScore());
+    messageEntity.setProperty("parentKey", message.getParentKey());
```

Comment:
We shouldn't do it this way. Parent key should be initiated when creating the Entity: https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Java_Ancestor_paths

Reply:
A new PR #33 is up to change the implementation. Please review there.